[virt_autotest] Add call parent post_fail_hook to virtual network test

guoxuguang opened the pull request:
Refer to the failure job https://openqa.suse.de/tests/4840195, which fail at isolated network test, two main issues:
- no parent post_fail_hook is triggered
- although in the test pm file , it shows failed as https://openqa.suse.de/tests/4840195#step/libvirt_isolated_virtual_network/113, but the junit log does not show its info at all

So, need to add call parent post_fail_hook to the following virtual network test

                tests/virt_autotest/libvirt_virtual_network_init
                tests/virt_autotest/libvirt_host_bridge_virtual_network
                tests/virt_autotest/libvirt_nated_virtual_network
                tests/virt_autotest/libvirt_routed_virtual_network
                tests/virt_autotest/libvirt_isolated_virtual_network

- Verification run: 
[gi-guest_developing-on-host_sles15sp2-kvm](http://149.44.176.58/tests/4912208)
[gi-guest_developing-on-host_sles15sp2-xen](http://149.44.176.58/tests/4913844)

guoxuguang commented:
@alice-suse @Julie-CAO @waynechen55 please help me review this PR.

Julie-CAO commented:
LGTM

alice-suse commented on tests/virt_autotest/libvirt_host_bridge_virtual_network.pm:
```diff
@@ -107,6 +107,7 @@ sub post_fail_hook {
 
     #Restore Network setting
     virt_autotest::virtual_network_utils::restore_network($virt_host_bridge, $based_guest_dir);
+    $self->SUPER::post_fail_hook;
```
Thanks for the effort! 

With calling SUPER::post_fail_hook as last step, which is after restart_libvirtd, destroy_virt_network, restore_guests, etc, many important instant state of host and guest will be lost when collecting host and guest supportconfig. So I am afraid it needs to be called immediately after test fail as first step in post_fail_hook, IF it does not affect later lines from L93~L109. One important impact of it will be to virt_autotest::virtual_network_utils::upload_debug_log(). 
- Please @guoxuguang help evaluate if the uploaded logs of this step is still valid.
- Besides, what logs are duplicated between virt_autotest::virtual_network_utils::upload_debug_log() and SUPER::post_fail_hook. Anything can be optimized here to avoid redundancy?

guoxuguang commented:
sure, had updated to call `$self->SUPER::post_fail_hook` immediately after test fail as first step in post_fail_hook.

Also, had confirm that `$self->SUPER::post_fail_hook` had covered `virt_autotest::virtual_network_utils::upload_debug_log()`.

So, removed `virt_autotest::virtual_network_utils::upload_debug_log() ` now

alice-suse commented:
LGTM

alice-suse commented:
@guoxuguang I guess you are doing new validation tests, right? Please give a msg when it is done and updated here.

guoxuguang commented:
> @guoxuguang I guess you are doing new validation tests, right? Please give a msg when it is done and updated here.

sure, I am doing a new validation tests for this PR. 
After it was done and updated here.

guoxuguang commented:
@alice-suse 
Add the latest Validation run:
[gi-guest_developing-on-host_sles15sp2-kvm](http://149.44.176.58/tests/4912208)

alice-suse commented:
Can you add a xen verification? I am thinking the logs in virt_autotest::virtual_network_utils::upload_debug_log() may not be fully covered in virt_logs_all.

guoxuguang commented:
> Can you add a xen verification? I am thinking the logs in virt_autotest::virtual_network_utils::upload_debug_log() may not be fully covered in virt_logs_all.

sure,  working on adding a xen verification now.

guoxuguang commented:
@alice-suse 
Add the latest xen validation run for this PR. 
[gi-guest_developing-on-host_sles15sp2-xen](http://149.44.176.58/t4913844)

If there was not any comment, if this PR is ready to merge. thanks.

alice-suse commented:
> > Can you add a xen verification? I am thinking the logs in virt_autotest::virtual_network_utils::upload_debug_log() may not be fully covered in virt_logs_all.
> 
> sure, working on adding a xen verification now.
> [gi-guest_developing-on-host_sles15sp2-xen](http://149.44.176.58/t4913844)

Thanks! From this validation job http://149.44.176.58/t4913844's uploaded virt_logs_all.tar.gz, in the host supportconfig , there are /var/log/xen and /var/log/libvirt(libxl-driver is also there) , and host boot log. So the virt_autotest::virtual_network_utils::upload_debug_log() can be safely removed IMO.

guoxuguang commented:
> > > Can you add a xen verification? I am thinking the logs in virt_autotest::virtual_network_utils::upload_debug_log() may not be fully covered in virt_logs_all.
> > 
> > 
> > sure, working on adding a xen verification now.
> > [gi-guest_developing-on-host_sles15sp2-xen](http://149.44.176.58/t4913844)
> 
> Thanks! From this validation job http://149.44.176.58/t4913844's uploaded virt_logs_all.tar.gz, in the host supportconfig , there are /var/log/xen and /var/log/libvirt(libxl-driver is also there) , and host boot log. So the virt_autotest::virtual_network_utils::upload_debug_log() can be safely removed IMO.

sure, agree on. 
Thank you so much for your great help.
